components/automate-ui/src/app/pages/team-details/team-details.component.ts

```diff
@@ -103,6 +97,21 @@ export class TeamDetailsComponent implements OnInit, OnDestroy {
   }
 
   ngOnInit(): void {
+    combineLatest([
+      this.store.select(routeParams),
+      this.store.select(routeURL)
+    ]).pipe(
+      takeUntil(this.isDestroyed),
+      map(([params, url]) => [params.id as string, url]),
+      // Only fetch if we are on the team details route, otherwise
+      // we'll trigger GetTeam with the wrong input on any route
+      // away to a page that also uses the :id param.
+      filter(([id, url]) => TEAM_DETAILS_ROUTE.test(url) && id !== undefined),
+      // Remove the url because we only need to check if the id has changed
+      map(([id, _url]) => id),
+      distinctUntilChanged()
+    ).subscribe(id => this.store.dispatch(new GetTeam({ id })));
```

Comment:
the GET action dispatch must be the first call on init

Reply:
Moving this block up as a benefit for human readers is fine, but note that because these are all asynchronous setup calls the order does not matter. If you are seeing some difference that may be hinting at a race condition that could need attention.